packages/storage/amplify_storage_s3/example/lib/main.dart

```diff
@@ -105,19 +108,271 @@ class HomeScreen extends StatefulWidget {
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.debug('Download error - ${e.message}');
+    }
+  }
+
+  // download file on web
+  Future<void> downloadFileWeb(String key) async {
+    try {
+      final result = await Amplify.Storage.downloadFile(
+        key: key,
+        localFile: AWSFile.fromPath(key),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.debug('Download error - ${e.message}');
+    }
+  }
+
+  // delete file from S3 bucket
+  Future<void> removeFile({
+    required String key,
+    required StorageAccessLevel accessLevel,
+  }) async {
+    try {
+      final result = await Amplify.Storage.remove(
+        key: key,
+        options: StorageRemoveOptions(accessLevel: accessLevel),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.debug('Delete error - ${e.message}');
+      rethrow;
+    }
+  }
+
+  // get the url of a file in the S3 bucket
+  Future<String> getUrl({
+    required String key,
+    required StorageAccessLevel accessLevel,
+  }) async {
+    try {
+      final result = await Amplify.Storage.getUrl(
+        key: key,
+        options: StorageGetUrlOptions(
+          accessLevel: accessLevel,
+          pluginOptions: const S3GetUrlPluginOptions(
+            validateObjectExistence: true,
+            expiresIn: Duration(minutes: 1),
+          ),
+        ),
+      ).result;
+      setState(() {
+        imageUrl = result.url.toString();
+      });
+      return result.url.toString();
+    } on StorageException catch (e) {
+      _logger.debug('Get URL error - ${e.message}');
+      rethrow;
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: const Text('Amplify Storage Example'),
       ),
-      body: const Center(
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            Text('Examples are coming...'),
-          ],
-        ),
+      body: Stack(
+        children: [
+          Center(
+            child: Padding(
+              padding: const EdgeInsets.all(20),
+              child: ListView(
+                children: [
+                  for (var item in list)
+                    ListTile(
+                      onTap: () => {
+                        getUrl(
+                          key: item.key as String,
+                          accessLevel: StorageAccessLevel.guest,
+                        )
+                      },
+                      title: Text(item.key as String),
+                      trailing: Row(
+                        mainAxisSize: MainAxisSize.min,
+                        children: [
+                          IconButton(
+                            icon: const Icon(Icons.delete),
+                            onPressed: () {
+                              removeFile(
```

Comment:
Looks like when I delete a file, but have it selected in the UI. The image URL and picture remain. We should probably clear UI on delete.